docs/python-integration.md

```diff
@@ -0,0 +1,302 @@
+        source_column="content",
+        embedding=EmbeddingConfig(
+            model="text-embedding-3-small",
+            dimensions=768
+        )
+    )
+```
+
+### Querying with VectorizerField
+
+Once your model is set up, you can query the embeddings in several ways:
+
+#### Basic ORM Queries
+
+```python
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+# Get all embeddings
+with Session(engine) as session:
+    # Get embedding entries
+    embeddings = session.query(BlogPost.content_embeddings).all()
+    
+    # Access embedding vectors and chunks
+    for embedding in embeddings:
+        vector = embedding.embedding  # numpy array of embeddings
+        chunk = embedding.chunk      # text chunk that was embedded
+        chunk_seq = embedding.chunk_seq  # sequence number of chunk
+```
+
+#### Semantic Search
+
+Using pgvector's distance operators in queries:
+
+```python
+from sqlalchemy import func
+
+def search_similar_content(session: Session, query_text: str, limit: int = 5):
+    # Generate embedding for search query
+    return (
+        session.query(
+            BlogPost.content_embeddings,
+            BlogPost.title,
+            # Calculate and include distance in results
+            BlogPost.content_embeddings.embedding.cosine_distance(
+                func.ai.openai_embed(
+                    'text-embedding-3-small',
+                    query_text,
+                    text('dimensions => 768')
+                )
+            ).label('distance')
+        )
+        .order_by('distance')  # Sort by similarity
+        .limit(limit)
+        .all()
+    )
+
+# Usage:
+results = search_similar_content(session, "machine learning concepts")
+for embedding, title, distance in results:
+    print(f"Title: {title}")
+    print(f"Matching chunk: {embedding.chunk}")
+    print(f"Distance: {distance}\n")
+```
+
+#### Advanced Filtering
+
+```python
+# Find content within a certain distance threshold
+threshold_query = (
+    session.query(BlogPost.content_embeddings)
+    .filter(
+        BlogPost.content_embeddings.embedding.cosine_distance(
+            func.ai.openai_embed(
+                'text-embedding-3-small',
+                'search query',
+                text('dimensions => 768')
+            )
+        ) < 0.3
+    )
+)
+
+# Combine with regular SQL filters
+combined_query = (
+    session.query(BlogPost, BlogPost.content_embeddings)
+    .join(
+        BlogPost.content_embeddings,
+        BlogPost.id == BlogPost.content_embeddings.id,
+    )
+    .filter(BlogPost.title.ilike("%Python%"))
+    .order_by(
+        BlogPost.content_embeddings.embedding.cosine_distance(
+            func.ai.openai_embed(
+                'text-embedding-3-small',
+                'search query',
+                text('dimensions => 768')
+            )
+        )
+    )
+)
+```
+
+### Model Relationships
+
+You can optionally create SQLAlchemy relationships between your model and its embeddings:
```

Comment:
Would it be worth it to mention that we expose the relationship via the view that's created. I don't know much about sqlalchemy, so maybe this `relationship` way is better than querying the view directly.

Reply:
I need to look up exactly how relationships work, but generally speaking it's just an eager join that enables the related objects to be loaded as a list on the parent.
Mat mentioned that he's, understandably a bit skeptical about automatic join, because you can run into N+1 problems depending on how it is configured, which is why I made it optional.

Reply:
I wonder if we can set up a relationship but not have it do the eager join by default?